Server/Ebenezer/User.cpp

```diff
@@ -8690,6 +8690,16 @@ int CUser::GetNumberOfEmptySlots() const
 	return emptySlotCount;
 }
 
+bool CUser::CheckExistEvent(int questId, int questState) const
+{
+	for (const auto& quest : m_pUserData->m_quests)
+	{
+		if (quest.sQuestID == questId && quest.byQuestState == questState)
+			return true;
+	}
+	return false;
```

Comment:
It should return true at this point if we're requesting `QUEST_STATE_NOT_STARTED`, as there's no quest state for the associated quest yet.

Reply:
so the method should be like this then?
```c++
for (const auto& quest : m_pUserData->m_quests)
{
	if (quest.sQuestID != questId)
		continue;

	if (quest.byQuestState == questState)
		return true;
	break;
}
return questState == 0 ? true : false;
```

Reply:
Sorry, too used to too many different codebases. There probably isn't yet an enum defined for the quest states, but it should look something like:
```cpp
enum e_QuestState
{
    QUEST_STATE_NOT_STARTED = 0,
    QUEST_STATE_IN_PROGRESS = 1,
    QUEST_STATE_COMPLETE = 2
};
```

I would throw it in `shared/globals.h` as the client may have need for it.